DNN Platform/Library/Services/Exceptions/Exceptions.cs

```diff
@@ -305,14 +306,23 @@ public static void ProcessModuleLoadException(string FriendlyMessage, Control ct
                 {
                     var ctrlModule = ctrl as IModuleControl;
                     ModuleLoadException lex = null;
+                    var friendlyMessageOverride = exc.Message;
+
+                    if (Host.DebugMode && exc is HttpParseException)
+                    {
+                        var httpParseError = ((HttpParseException)exc);
+                        friendlyMessageOverride = string.Concat(exc.Message, " in ", httpParseError.VirtualPath, ":line ", httpParseError.Line);
+                    }
+
                     if (ctrlModule == null)
                     {
-                        lex = new ModuleLoadException(exc.Message, exc);
+                        lex = new ModuleLoadException(friendlyMessageOverride, exc);
```

Comment:
Yes, I wanted to improve the errors when we run into an HttpParseException (triggered when the namespace doesn't match the control registered in the ascx). Currently you need to attach the debugger to figure out which sub ascx has a parse exception. This change will update the friendly exception message and include the control and line number with the parse exception.